extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/binder/CommonQueryBinder.java

```diff
@@ -55,4 +57,27 @@ static String escape(Object value) {
         }
         return "'" + value.toString().replace("\\", "\\\\").replace("'", "\\'") + "'";
     }
+
+    /**
+     * Method used to apply in $in operators
+     */
+    private static String arrayValue(Object value) {
+        StringBuilder arrayValue = new StringBuilder();
+        List valueList = valueToList(value);
+        for (int i = 0; i < valueList.size(); i++) {
+            arrayValue.append(escape(valueList.get(i)));
+            if (i + 1 < valueList.size()) {
+                arrayValue.append(", ");
+            }
+        }
+        return arrayValue.toString();
+    }
+
+    private static List valueToList(Object value) {
+        if (value.getClass().isArray()) {
+            return Arrays.asList((Object[]) value);
+        }
+
+        return new ArrayList<>((Collection) value);
+    }
```

Comment:
You can use an iterator or turn you collection into an array to avoid creating a new ArrayList from the collection.
And you can use a `StringJoiner` to avoid dealing with leading `,` ...

Reply:
Done.

Reply:
Thanks, it's nicer ;)